scipy/spatial/ckdtree.pyx

```diff
@@ -601,6 +605,7 @@ cdef class cKDTree:
 
         compact = 1 if compact_nodes else 0
         median = 1 if balanced_tree else 0
+        workers = build_workers
```

Comment:
We have this logic someplace else. I think this would be better to just copy this (alternatively make it as a util function to be reused elsewhere).
```python
    workers = int(workers)
    if workers == -1:
        workers = os.cpu_count()  # type: ignore[assignment]
        if workers is None:
            raise NotImplementedError(
                "Cannot determine the number of cpus using os.cpu_count(), "
                "cannot use -1 for the number of workers"
            )
    elif workers <= 0:
        raise ValueError(f"Invalid number of workers: {workers}, must be -1 "
                         "or > 0")
```

Reply:
I did actually copy from here and adapted it, just missed the detail that `os.cpu_count()` may return `None`. I edited my suggestion to fix.